src/libs/Violations/ViolationsUtils.ts

```diff
@@ -126,30 +126,27 @@ const ViolationsUtils = {
             case 'missingComment':
                 return translate('violations.missingComment');
             case 'missingTag':
-                return translate('violations.missingTag', {tagName: violation.data?.tagName});
+                return translate('violations.missingTag', violation.data ?? undefined);
```

Comment:
If I understand correctly, by passing `violation.data` here, since the type for this is `type ViolationsMissingTagParams = {tagName?: string};`, object deconstructing happens automatically, so we don't have to explocitly pass `tagName`

But in some cases below, we explicitly pass `formattedLimit`. Can we make these consistent?

Reply:
Refactored this to gather all the destructuring and initialization at the top of the function.

There is still one case that I think we should address which is related to `autoReportedRejectedExpense`:

```
autoReportedRejectedExpense: ({rejectReason, rejectedBy}: ViolationsAutoReportedRejectedExpenseParams) => `${rejectedBy} rejected this expense with the comment "${rejectReason}"`,
```
```
autoReportedRejectedExpense: ({rejectedBy, rejectReason}: ViolationsAutoReportedRejectedExpenseParams) => `${rejectedBy} rechazó la solicitud y comentó "${rejectReason}"`,
```

While it should never happen, it's possible in the code as written for `rejectedBy` and `rejectReason` to be `undefined` -- because we can't guarantee that `violation.data` won't be `null`.

Are there default values that make sense for these?

Like maybe `rejectedBy` could be "The policy manager" by default, and we could make "with the comment"
 conditional 

So the strings would be like:

```
The policy administrator rejected this expense
The policy administrator rejected this expense with the comment "sorry"
John Smith rejected this expense
John Smith rejected this expense with the comment "sorry"
```

Reply:
No not really, so maybe to be safe we can just check if neither exists, and return `''`, and if both exist, we return the string we have in there

Reply:
```ts
            case 'autoReportedRejectedExpense':
                return rejectReason && rejectedBy
                    ? translate('violations.autoReportedRejectedExpense', {
                          rejectedBy,
                          rejectReason,
                      })
                    : '';
```